### DISCUSSION TITLE
MEC Application

### DISCUSSION THREAD

**Discussion by achi07** [InterDigitalInc/AdvantEDGE#377] [General]
Hi,

I have 3 VMs, one of them is hosting the AdvantEDGE platform in which the scenario is running. The scenario consists of 2 MEC hosts, where as the other 2 VMs are dedicated for edge1 and edge2. The edge1 host the mec application and the services. How to direct the traffic of mec app and services towards the AdvantEDGE platform, so that the UE reaches to the MEC App via AdvantEDGE platform?

Thank you.

**Comment by userH1213** [comment]
have you tried to set your UEs as externals. set the ports properly and hopefully that will make it work. you can take a look at 
[External nodes support](https://interdigitalinc.github.io/AdvantEDGE/docs/overview/features/ext-nodes/#external-node-support)

**Comment by achi07** [comment]
Hi, 

Yes, I have set the UE app as the external, as well as the edge app. I have followed the pattern mentioned in the link. the ingress for an external UE and the egress for an external edge app for this combination of External-UE to External Edge/Fog/cloud Server.

**Comment by dilallkx** [comment]
Hi @achi07,

Thank you for your question!

If I understand correctly, you have:
- AdvantEDGE running on VM#1
- an external UE application running on VM#2
- an external edge application running on VM#3

First, do you confirm that the UE application on VM#2 can communicate directly with the edge application running on VM#3? If so, I recommend capturing a network trace to obtain a list of all traffic flow ports required by your application.

To get the UE application to communicate with the edge application through AdvantEDGE you will need to do the following:
- Create a scenario with an external UE application & edge application
- Configure the necessary ingress & egress rules for the application traffic flows as described in the [documentation](https://interdigitalinc.github.io/AdvantEDGE/docs/overview/features/ext-nodes/).
- Configure your external applications to use the configured AdvantEDGE IP address & node ports
- Deploy the scenario & start your external applications

If you continue to observe issues, please capture network traces of your application traffic between and in your VMs. This should help identify if & where the traffic is being dropped or blocked and will help verify your scenario configuration.

Also, please note that we have had some issues with this type of deployment in the past, as documented in this [FAQ](https://interdigitalinc.github.io/AdvantEDGE/docs/project/project-faq/#faq-1-unexplained-disconnections-using-an-egress-service).

I hope this helps resolve your issue.
Regards!

**Comment by achi07** [comment]
Hi, Thank you so much for your response. I have the following configuration

- AdvantEDGE running on VM#1 (192.168.54.21)
- an external UE application running on VM#2 Ingress Mapping (31000:my-service:9000:TCP)
- an external edge application running on VM#3, Egress Mapping (my-service::192.168.54.17:9000:TCP)

As per the External Node Support documentation,  I did the the ingress for an external UE and the egress for an external edge app in the AdvantEDGE platform as mentioned above. The mec-app is a video application, for the external UE application I am using VLC app. To open the stream I am fetching the AdvantEDGE IP and the port which 192.168.54.21:31000.

I am receiving some data value in the Downlink (Grafana Dashboard), which is shown as 0.043 Mbps, the video quality is good there is no interrupt. But why the Downlink value is so low? Do I need to do the conversation? In the activity monitor of the VM2 I can see the receiving link value around 3Mbps. Also, in the mec-app and UE-app I have configured the value of 1000 for the UL, DL. 

Thank you so much, please let me know about this.

**Reply by dilallkx** [reply]
Hi @achi07,

Sorry for the delayed response!

Have you captured a network trace of your traffic? The above configuration should work for TCP traffic flows originating from the external UE and sent to the AdvantEDGE IP @ 192.168.54.21:31000 (then forwarded to the external service on port 9000). This seems to work for you, however the dashboard is not displaying the correct throughput. This could be due to:
- The dashboard filter may not be set to the correct application (e.g. external UE app or external edge app) -- this is unlikely.
- The VLC video stream may be sent using a new flow with different protocol or ports
  - Did you capture a network trace to see if the TCP stream for the video is initiated by the UE and continues to use port 9000?
  - Is it possible the data stream uses a different port?

I hope this helps you find the problem.
Regards!
